core/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -307,10 +307,16 @@ public static GetResult fromXContentEmbedded(XContentParser parser) throws IOExc
                         fields.put(getField.getName(), getField);
                     }
                 } else {
-                    throwUnknownField(currentFieldName, parser.getTokenLocation());
+                    parser.skipChildren(); // skip potential inner objects for forward compatibility
                 }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                parser.skipChildren(); // skip potential inner arrays for forward compatibility
             }
         }
+
+        if (found == null) {
```

Comment:
I'm wrong, at this stage and when parsing an UpdateResponse the index/type/id/version fields can be missing because they are printed out by the UpdateResponse itself...

I removed the checks from this method and add them in the GetResponse.fromXContent() where I think they should be located. This way we keep the same behavior (ie throwing an exception when fields are missing) but that would impact only users of the GetResponse.fromXContent() method.